beacon-chain/sync/initial-sync/service.go

```diff
@@ -239,3 +198,43 @@ func (s *Service) waitForMinimumPeers() {
 		time.Sleep(handshakePollingInterval)
 	}
 }
+
+// waitForStateInitialization makes sure that beacon node is ready to be accessed: it is either
+// already properly configured or system waits up until state initialized event is triggered.
+func (s *Service) waitForStateInitialization() time.Time {
+	var genesis time.Time
+	headState, err := s.chain.HeadState(s.ctx)
+	if headState == nil || err != nil {
+		log.Info("Waiting for state to be initialized")
+		// Wait for state to be initialized.
+		stateChannel := make(chan *feed.Event, 1)
+		stateSub := s.stateNotifier.StateFeed().Subscribe(stateChannel)
+		genesisSet := false
+	loop:
```

Comment:
we dont really use control statements like this in the repo, I dont see any advantage/necessity having `loop:` here.